language/move-vm/runtime/src/interpreter.rs

```diff
@@ -101,15 +102,45 @@ impl Interpreter {
         extensions: &mut NativeContextExtensions,
         loader: &Loader,
     ) -> VMResult<Vec<Value>> {
-        Interpreter {
+        let mut interpreter = Interpreter {
             operand_stack: Stack::new(),
             call_stack: CallStack::new(),
             paranoid_type_checks: loader.vm_config().paranoid_type_checks,
             runtime_limits_config: loader.vm_config().runtime_limits_config.clone(),
+        };
+        if function.is_native() {
+            for arg in args {
+                interpreter
+                    .operand_stack
+                    .push(arg)
+                    .map_err(|e| e.finish(Location::Undefined))?;
+            }
+            let resolver = function.get_resolver(loader);
+            let return_values = interpreter
+                .call_native_return_values(
```

Comment:
just out of curiosity... I have a PR where I moved collecting the return values from the stack in this function.
If we had that we could have called directly the `call_native` as it was and fetch the value from the stack... or not?

Reply:
Ah, yeah we could retrieve them from the stack. Do you want me to attempt to do that in this PR?